### ISSUE TITLE
Fatal error: Unexpected Token <

### ISSUE THREAD

**Issue by mwmwmw** [signalkuppe/grunt-gathercontent#1]
This seems to only happen on my Mac (windows works, will double check when I get back to my desk machine). I'm not sure what the issue is, whether its the gather content task, or the 

Running "gathercontent-get_projects" task
Fatal error: Unexpected token <

A few of the tasks work... these all return their respective JSON files.

get_files_by_pages
get_custom_states_by_project
get_project
get_pages

downloadfiles finishes, but doesn't download any files.

**Comment by signalkuppe** [comment]
Have you already checked on your desk machine?
I've just ran the gathercontent-get_projects task on my env and everything is fine.
Can you give more info about the error?

**Comment by mwmwmw** [comment]
Hey, sorry for the delay, yes it's only on my Mac that gives the error. 

on my windows machine, works fine. The code is identical. I'm running the grunt task in Webstorm both times. 

I just had a thought.. maybe it's permissions in the terminal in OSX.

**Comment by mwmwmw** [comment]
Alright, we got to the bottom of this finally. 

What GC is asking for some kind of "length" parameter in the request. When GC doesn't see this parameter it responds with an HTML file and a "411 Length Required" error. 

The fix is really easy. Starting on Line 37 of gathercontent.js

``` javascript
        headers: {
          "Content-Type":"application/x-www-form-urlencoded",
          "Content-Length": 0 // <-- added to defaultOptions
        }

```

**Comment by signalkuppe** [comment]
Hi Matt, now i can reproduce it to, since 0.4 api version the task throws this
error.
I'm going to fix it and merge your fix with file downloads

thanks again!

On Fri, Aug 14, 2015 at 12:12 AM, Matthew Willox notifications@github.com
wrote:

> Closed #1 https://github.com/signalkuppe/grunt-gathercontent/issues/1.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/signalkuppe/grunt-gathercontent/issues/1#event-381994906
> .

## 

Matteo Leoni --

http://www.signalkuppe.com

**Comment by mwmwmw** [comment]
Hey, no problem! 

That was my suspicion as well, that it was something that changed in the API.

I don't think that setting the content-length to 0 ended up being a good fix. More of a hack. It fixed some things, but broke others.
